src/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -777,25 +777,19 @@ public static void TestIsWellFormedOriginalString()
             Assert.False(b);
         }
 
-        [Fact]
-        public static void TestIsWellFormedUriString()
+        [Theory]
+        [InlineData("http://www.contoso.com/path?name", true)]
+        [InlineData("http://www.contoso.com/path???/file name", false)]
+        [InlineData("c:\\directory\filename", false)]
+        [InlineData("file://c:/directory/filename", false)]
+        [InlineData("http:\\host/path/file", false)]
+        [InlineData("http://www.contoso.com/ sek� http://test.com", false)]
```

Comment:
Just two comments on these test cases:

(1) We typically represent Unicode characters with the '\u' syntax. So this line would become:
```csharp
[InlineData("http://www.contoso.com/ sek\u00E4 http://test.com", false)]
```
(2) we should add some cases that cover valid URIs containing Unicode e.g,
```csharp
[InlineData("http://www.contoso.com/\u00E4", true)]
[InlineData("http://www.contos\u00E4.com/path", true)]
```

Reply:
good catch thank's for this sample `[InlineData("http://www.contoso.com/\u00E4", true)]` now this fail, seems there aren't other tests like this on code base. Go on with investigation for fix.

Reply:
```cs
        public static IEnumerable<object[]> TestIsWellFormedUriStringData =>
        new List<object[]>
        {
            new object[] { "http://www.contoso.com/path?name", true },
            new object[] { "http://www.contoso.com/path?name ", true },
            new object[] { "http://www.contoso.com/path? name ", false },

            new object[] { "http://www.contoso.com/path?\u00E4", true },
            new object[] { "http://www.contoso.com/path?\u00E4 ", true },
            new object[] { "http://www.contoso.com/path? \u00E4 ", false },

            new object[] { "http://www.contoso.com/path?p=", true },
            new object[] { "http://www.contoso.com/path?\u00E4=", true },
            new object[] { "http://www.contoso.com/path?p= ", true },
            new object[] { "http://www.contoso.com/path?\u00E4= ", true },
            new object[] { "http://www.contoso.com/path?p= val", false },
            new object[] { "http://www.contoso.com/path?\u00E4= \u00E4", false },

            new object[] { "http://www.contoso.com/path???/file name", false },
            new object[] { "c:\\directory\filename", false },
            new object[] { "file://c:/directory/filename", false },
            new object[] { "http:\\host/path/file", false },

            new object[] { "http://www.contoso.com/ sek\u00E4 http://test.com", false },
            new object[] { "http://www.contoso.com/ \u00E4 sek http://test.com", false },
            new object[] { "http://www.contoso.com/\u00E4/sek http://test.com", false },
            new object[] { "http://www.contoso.com/a/sek http://test.com", false },
            new object[] { "http://www.contoso.com/ sek http://test.com", false },

            new object[] { "http://www.contoso.com/\u00E4/", true },
            new object[] { "http://www.contos\u00E4.com/path", true },
            new object[] { "http://www.contos\u00E4.com/path ", false }
        };
```
@rmkerr i've added more tests to be sure to cover all cases, can you check if my expetation are correct and/or add others...after some debugging my impression is that the issue is also related to the position of unicode chars. Thank's for help.

Reply:
FYI Working on private branch https://github.com/dotnet/corefx/compare/master...MarcoRossignoli:iwfuridev

Reply:
Great, thanks for adding all of those test cases!